src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -14,70 +14,168 @@
 
-  /**
-   * Build and return an authorized API client service.
-   *
-   * @return an authorized API client service
-   * @throws GeneralSecurityException, IOException
-   */
-  public static YouTube getService() throws GeneralSecurityException, IOException {
-    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
-        .setApplicationName(APPLICATION_NAME)
-        .build();
+  @Before
+  public void setUp() {
+    topCommentSnippet.setTextDisplay("Test Message");
+    testTopComment.setSnippet(topCommentSnippet);
+    testThreadSnippet.setTopLevelComment(testTopComment);
+    testCommentThread.setSnippet(testThreadSnippet);
+    youtubeResponse.setItems(testCommentThreadList);
   }
 
-  @Before
-  public void setUp() throws GeneralSecurityException, IOException {
-    YouTube youtubeService = getService();
-    YouTube.CommentThreads.List youtuberequest = youtubeService.commentThreads().list(SNIPPET);
-    youtuberesponse =
-        youtuberequest
-            .setKey(DEVELOPER_KEY)
-            .setVideoId(TEST_VIDEO_ID)
-            .setMaxResults(2L)
-            .setTextFormat(PLAINTEXT)
-            .execute();
-    analysis = new CommentAnalysis();
+  @Rule public ExpectedException exception = ExpectedException.none();
+
+  @Test
+  public void testCalculateSentiment() {
+    // This is a test method to calculate simulate and test the process in comment analysis language
+    // service
+    when(mockedlanguageService
+            .analyzeSentiment(any(Document.class))
+            .getDocumentSentiment()
+            .getScore())
+        .thenReturn(TEST_SCORE);
+    Statistics testStat = commentAnalysis.computeOverallStats(youtubeResponse);
+    Assert.assertNotNull(testStat);
+    Assert.assertNotNull(testStat.getAggregateValues());
+    Assert.assertEquals(testStat.getAverageScore(), 0.23, 0.01);
+    Assert.assertEquals(
+        testStat
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(0.2), BigDecimal.valueOf(0.4)))
+            .intValue(),
+        2);
+    Assert.assertEquals(
+        testStat
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(0.2), BigDecimal.valueOf(0.4)))
+            .intValue(),
+        2);
+  }
+
+  @Test
+  public void testNormalSymmetricCases() {
+    Statistics normalStat = new Statistics(SCORE_IN_RANGE);
+    Statistics symmetricStat = new Statistics(SYMMETRIC_SCORE);
+    Assert.assertEquals(
+        normalStat
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(0.0), BigDecimal.valueOf(0.2)))
+            .intValue(),
+        4);
+    Assert.assertEquals(
+        normalStat
+            .getAggregateValues()
+            .get(new Range(BigDecimal.valueOf(-0.2), BigDecimal.valueOf(0.0)))
+            .intValue(),
+        2);
+    Assert.assertTrue(Math.abs(normalStat.getAverageScore()) <= 1);
```

Comment:
Since you are already testing the averageScore below, you may want to remove this as this adds no extra information.